src/xml_output.rs

```diff
@@ -8,15 +8,15 @@ use xml::{writer::XmlEvent, EmitterConfig, EventWriter};
 
 use failure::{bail, format_err, Error};
 
-pub trait BinXmlOutput<'a, W: Write> {
+pub trait BinXmlOutput<W: Write> {
     fn with_writer(target: W) -> Self
     where
         Self: Sized;
     fn into_writer(self) -> Result<W, Error>;
     fn into_writer_from_box(self: Box<Self>) -> Result<W, Error>;
 
     fn visit_end_of_stream(&mut self) -> Result<(), Error>;
-    fn visit_open_start_element(
+    fn visit_open_start_element<'a>(
```

Comment:
why did you scope the lifetime of the trait? this would mean that we would need to add `<'a>` locally to all functions that may contain data referencing the tokens. Isn't this the same as having `'a` at the trait level?